### ISSUE TITLE
Execution Housekeeping

### ISSUE THREAD

**Issue by wdbaruni** [bacalhau-project/bacalhau#3857]
## The Problem
The way we communicate executions to the compute nodes today is through a `ComputeForwarder` Planner that picks up plan by the scheduler and call compute node APIs. This has limitations that when a requester node restarts, or when the forwarded fails to call a compute node, the execution will be in a stuck state. Also it has a limitation that we wont get status updates or confirmations that the execution is still active in the compute node.

## The Proposal
With the introduction of [Heartbeat execution progress ](https://github.com/bacalhau-project/bacalhau/issues/3859) that will update the executions actual state, and we already have the schedulers that update the executions desired state, the proposal is to have a background controller that pulls the executions state and:
1. Check for new desired state changes by the scheduler and forwards them to the compute nodes
2. Check for actual state updates by the [Heartbeat execution progress ](https://github.com/bacalhau-project/bacalhau/issues/3859), and enqueue new evaluations if necessary 

In summary, the Housekeeper is a background task the keeps comparing the desired and actual states, and takes actions accordingly. This provides a more resilient management of the executions and covers more edge cases compared to the current `ComputeForwarder`. We might keep the `ComputeFrowarder` as well to faster communications, but a background controller is a must
